### PR TITLE
LibGfx/JPEG: Allow decoding more subsampling factors 

### PR DESCRIPTION
**Pull request by nico**
We now allow all subsampling factors where the subsampling factors
of follow-on components evenly decode the ones of the first component.

In practice, this allows YCCK 2111, CMYK 2112, and CMYK 2111.

### CODE REVIEW

**Comment by LucasChollet** [comment]
Thanks for taking care of that (and sorry that you get bored of waiting for my patch)!

I'll review that later!

**Comment by nico** [comment]
I dropped an unrelated simplification that I'll put in a separate PR once this is in.

With this, all chroma subsampling issues in my 1000 test PDFs disappear (…even though I think we still don't support everything listed on https://calendar.perfplanet.com/2015/why-arent-your-images-using-chroma-subsampling/ ).

Before:

Screenshot 2024-01-14 at 7 29 38 PM

After:

Screenshot 2024-01-14 at 7 33 16 PM

Before:

Screenshot 2024-01-14 at 7 30 02 PM

After:

Screenshot 2024-01-14 at 7 33 24 PM

Before:

Screenshot 2024-01-14 at 7 30 39 PM

After (Preview.app / Safari can't display this image!):

Screenshot 2024-01-14 at 7 34 22 PM

Before:

Screenshot 2024-01-14 at 7 31 32 PM

After:

Screenshot 2024-01-14 at 7 36 02 PM

Before:

Screenshot 2024-01-14 at 7 32 33 PM

After:

Screenshot 2024-01-14 at 7 36 15 PM

File: Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp [hunk 1]
```diff
@@ -1589,8 +1586,7 @@ static void inverse_dct(JPEGLoadingContext const& context, Vector<Macroblock>& m
 static void undo_subsampling(JPEGLoadingContext const& context, Vector<Macroblock>& macroblocks)
 {
     // The first component has sampling factors of context.sampling_factors, while the others
-    // are either 1x1 or for the 4th component also context.sampling_factors. See
-    // read_start_of_frame() which currently enforces these restrictions.
+    // are divide the first component's sampling factors. This is enforced by read_start_of_frame().
```
Comment (LucasChollet) [inline Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp:1589]: Is it me not understanding English? "are divide" doesn't seem correct.

Comment (nico) [reply Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp]: That was just wrong, thanks :)

**Review by LucasChollet — APPROVED** [review]

**Pull request merged**